Replace deprecated test android classes InstrumentationRegistry and ActivityTestRule

Apparently this is happening continually, but only in certain cases does it fail the build. For example, [this run on master](https://btx.cloud.google.com/invocations/979814f3-6896-47c5-b45e-40b4beb796d2) passed, but has these warnings.

```
> Task :grpc-android-interop-testing:compileDebugAndroidTestJavaWithJavac FAILED
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:22: warning: [deprecation] InstrumentationRegistry in androidx.test has been deprecated
import androidx.test.InstrumentationRegistry;
                    ^
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:24: warning: [deprecation] ActivityTestRule in androidx.test.rule has been deprecated
import androidx.test.rule.ActivityTestRule;
                         ^
error: warnings found and -Werror specified
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:63: warning: [deprecation] ActivityTestRule in androidx.test.rule has been deprecated
  public ActivityTestRule<TesterActivity> activityRule =
         ^
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:64: warning: [deprecation] ActivityTestRule in androidx.test.rule has been deprecated
      new ActivityTestRule<TesterActivity>(TesterActivity.class);
          ^
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:68: warning: [deprecation] InstrumentationRegistry in androidx.test has been deprecated
    testCase = InstrumentationRegistry.getArguments().getString("test_case", "all");
               ^
grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:68: warning: [deprecation] getArguments() in InstrumentationRegistry has been deprecated
    testCase = InstrumentationRegistry.getArguments().getString("test_case", "all");
                                      ^
1 error
6 warnings

[Incubating] Problems report is available at: file://grpc-java/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':grpc-android-interop-testing:compileDebugAndroidTestJavaWithJavac'.
> Compilation failed; see the compiler output below.
  grpc-java/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java:22: warning: [deprecation] InstrumentationRegistry in androidx.test has been deprecated
  import androidx.test.InstrumentationRegistry;
                      ^
  1 error
  6 warnings
```

--- a/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java
+++ b/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java
@@ -19,9 +19,9 @@ package io.grpc.android.integrationtest;
 import static org.junit.Assert.assertEquals;
 
 import android.net.LocalSocketAddress.Namespace;
-import androidx.test.InstrumentationRegistry;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
-import androidx.test.rule.ActivityTestRule;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.platform.app.InstrumentationRegistry;
 import io.grpc.Grpc;
 import io.grpc.InsecureServerCredentials;
 import io.grpc.Server;
@@ -60,8 +60,8 @@ public class UdsChannelInteropTest {
   // Ensures Looper is initialized for tests running on API level 15. Otherwise instantiating an
   // AsyncTask throws an exception.
   @Rule
-  public ActivityTestRule<TesterActivity> activityRule =
-      new ActivityTestRule<TesterActivity>(TesterActivity.class);
+  public ActivityScenarioRule<TesterActivity> activityRule =
+      new ActivityScenarioRule<>(TesterActivity.class);
 
   @Before
   public void setUp() throws IOException {

--- a/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java
+++ b/android-interop-testing/src/androidTest/java/io/grpc/android/integrationtest/UdsChannelInteropTest.java
@@ -19,8 +19,8 @@ package io.grpc.android.integrationtest;
 import static org.junit.Assert.assertEquals;
 
 import android.net.LocalSocketAddress.Namespace;
-import androidx.test.ext.junit.runners.AndroidJUnit4;
 import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
 import androidx.test.platform.app.InstrumentationRegistry;
 import io.grpc.Grpc;
 import io.grpc.InsecureServerCredentials;

